compressors/fano_coder.py

```diff
@@ -0,0 +1,160 @@
+"""Fano Coding
+Shannon coding is a basic prefix free coding. There is some confusion in literature about Shannon, Fano and
+Shannon-Fano codes, e.g. see Naming section of https://en.wikipedia.org/wiki/Shannon–Fano_coding.
+
+This document uses Fano coding as described in the wiki article above.
+"""
+from typing import Any, Tuple
+from utils.bitarray_utils import BitArray
+from utils.test_utils import get_random_data_block, try_lossless_compression
+from compressors.prefix_free_compressors import PrefixFreeTree, PrefixFreeEncoder, PrefixFreeDecoder, BinaryNode
+from core.prob_dist import ProbabilityDist
+
+
+class FanoTree(PrefixFreeTree):
+    def __init__(self, prob_dist):
+        super().__init__(prob_dist)
```

Comment:
sorry, what's this doing?

Reply:
I see - I think I understand what's happening. Is doing stuff to super like this considered good practice even when we don't know what super is? Perhaps better is to just set to self here and have the super access these like `self.tree.prob_dist`?

Reply:
More generally, I think for any fancy python features we use - let's add a link to docs so people can read up. I am sure there are others like me who haven't had formal Python training struggle with some of the advanced concepts. I don't think that should come in the way of someone focused on understanding compression concepts.

Reply:
super is a standard way to initialize variable of a base class in an inherited class. We know in this case base class requires prob_dist which our inherited class also requires. We can add a comment to the doc: https://stackoverflow.com/questions/576169/understanding-python-super-with-init-methods

Would this work?

Reply:
Yeah, I think super() is a reasonably commonly used feature. @shubhamchandak94: For single inheritance, super() is just the parent class.. I believe we have this quite a bit in our codebase already